server/src/main/java/com/tencent/rss/server/ShuffleServerFactory.java

```diff
@@ -41,7 +41,7 @@ public ServerInterface getServer() {
     }
   }
 
-  enum ServerType {
+  protected enum ServerType {
```

Comment:
why protected?

Reply:
Accessed by the MockedShuffleServerFactory

Reply:
ServerType be accessible for its inheritors. @jerqi

Reply:
default accessible, put the classes into one pacakge.

Reply:
Moved it to server/test